### ISSUE TITLE
[JENKINS-53462] Jenkins websites use non-trusted 'submit' event to start form submission when current browser is Firefox

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#3478]
<p>HTML spec [<span class="error">&#91;1&#93;</span>|<a href="https://w3c.github.io/uievents/#trusted-events" class="external-link" target="_blank" rel="nofollow noopener">https://w3c.github.io/uievents/#trusted-events</a>] says "Most untrusted events will not trigger default actions, with the exception of the click event.". Now Firefox doesn't comply with the spec. When I try to fix the bug [<span class="error">&#91;2&#93;</span>|<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1370630</a>], a regression has happened on all Jenkins websites. Users can't login Jenkins websites with Firefox anymore. After some experiments, it seems the Jenkins websites detect the browser's user agent and use untrusted 'submit' event to start form submission when the current browser is Firefox. Changing the UA of Chrome to the same string as Firefox also block the form submission.</p>

<p> </p>

<p>The steps I used to reproduce this problem</p>

<p>On Chrome</p>
<ol>
	<li>Change UA to the same string as Firefox</li>
	<li>Navigate <a href="https://jenkins.qa.ubuntu.com/" class="external-link" target="_blank" rel="nofollow noopener">https://jenkins.qa.ubuntu.com/</a></li>
	<li>Click login</li>
	<li>Enter username/password and press 'log in' button</li>
	<li>Nothing happened</li>
</ol>

<p>Expectation</p>

<p>Don't use untrusted events to start form submission on Jenkins websites.</p>

<p> </p>

<p><span class="error">&#91;1&#93;</span> <a href="https://w3c.github.io/uievents/#trusted-events" class="external-link" target="_blank" rel="nofollow noopener">https://w3c.github.io/uievents/#trusted-events</a></p>

<p><span class="error">&#91;2&#93;</span> <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1370630</a></p>

<p> </p>

---
<details><summary><i>Originally reported by <a title="ming-chou shih" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=iamstone">iamstone</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-53462" target="_blank">Jenkins websites use non-trusted 'submit' event to start form submission when current browser is Firefox</a></i></summary>
<i><ul>
<li><b>reported by</b>: iamstone
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2021-04-20T13:29:16+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 9
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rtyler">rtyler</a>:</i>
<p>The WEBSITE project is for jenkins.io and those sorts of issues.</p>

<p> </p>

<p>That said, jenkins.qa.ubuntu.com is running a version of Jenkins more than two years old at this point.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=top12345tw">top12345tw</a>:</i>
<p>Hi, I could not reproduce this problem in the weekly version of Jenkins, but I still could reproduce it on LTS version. Should we reopen this issue for LTS version? Thank you.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>Let me quickly summarize. There is a <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630" class="external-link" target="_blank" rel="nofollow noopener">bug in Firefox</a> which we want to get fixed. But that is not possible because with the patched version of Firefox a login it not possible anymore due to the usage of untrusted submit events.</p>

<p> </p>

<p>To actually reproduce the bug you cannot use a normal release of Firefox but it would have to be one from the "first release with" here: <a href="https://hg.mozilla.org/mozilla-central/rev/367b6f947f87." class="external-link" target="_blank" rel="nofollow noopener">https://hg.mozilla.org/mozilla-central/rev/367b6f947f87.</a></p>

<p> </p>

<p>I'm fairly sure that waiting a week is totally fine if the above referenced fix in Jenkins is really the one which we need. It was just a speculation because I cannot test it myself.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>OK. The bug currently in Firefox allows the login to happen, which is why I never encountered the described problem. Once the bug in Firefox is fixed, it will no longer be able to use the login form. Now I got it.</p>

<p>Would be interesting to know whether simply giving Firefox the 'Chrome behavior' (if any), i.e. modifying or removing browser detection, would make this work. That should be fairly straightforward, I hope.</p>

<p>I'll move this to the correct Jira project.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> Can you clarify what release of Firefox this should be in? From my reading, it should be in 56.x, but I'm on 61, soon 62, and can log in just fine, current or previous login form.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>We had to backout the patch due to the regression with Jenkins. As such it was only available in some Nightly builds of Firefox during the 56 development process.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>Ok, so I checked with such a Nightly build and switched the user agent to Chrome. That actually fixed it, yes.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>I wonder what happens with Jenkins in current releases of Firefox if we remove the special treatment. I assume that was added for a reason.</p>

<p>IOW, do we rip out the Firefox special behavior, or do we need more elaborate Firefox special behavior to remain compatible with current and future Firefox.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=top12345tw">top12345tw</a>:</i>
<p>I checked the LTS 2.121.2 and 2.121.3, I still saw the Firefox special behavior. I am not sure if 2.138.1 fixes this, but I am happy to test again.</p>

<p>From the form submission point of view, it is not necessary to give Firefox special behavior, giving the same behavior as other browser should just also work on Firefox.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Should be testable by setting a different User-Agent on a prod release of Firefox I suppose.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>And assuming Chrome behaves to spec (like Firefox post-fix), we should be able to observe the problem in Jenkins by setting Chrome's User-Agent to Firefox.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=top12345tw">top12345tw</a>:</i>
<p>Or you can test this by setting User-Agent to Firefox on Chrome.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<blockquote><p>And assuming Chrome behaves to spec (like Firefox post-fix), we should be able to observe the problem in Jenkins by setting Chrome's User-Agent to Firefox.</p></blockquote>
<p>This is exactly what's happening.</p>

<p>The good news is that the redesigned login form will work in 2.138.1.</p>

<p>The bad news is that all other submit buttons will not. It's not quite as bad as before, as most forms have an 'Apply' button that still works, but close.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>Regarding the bad news, about how many different submit buttons we are talking here? Is it something which could easily be fixed and shipped, or how long would it take? So far Jenkins seems to be the only site which blocks us from shipping this feature fix, and we really would like to get this out to our users for a better web platform conformance.</p>

<p> </p>

<p>Thanks.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>I suspect it'll be one button, but it's reused a lot as the <tt>f:submit</tt> form control. Just needs someone to figure out where JS takes a wrong turn to address this.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler">tscherler</a>:</i>
<p>I will have a look now whether I can come up with a fix.</p>

<p>My understanding so far:</p>
<ul class="alternate" type="square">
	<li>I need a patched FF version (the earlier links do not work anymore, can you <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> provide me with a link?)</li>
	<li>The "new login" page works, as <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> said, most likely because I only used vanilla js, no external libs and especially no 
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">f:submit</pre>
</div></div></li>
	<li>in the bugzilla thread someone mentioned YUI I think that may be the base problem.</li>
</ul>


<p>I will now review the </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">f:submit</pre>
</div></div>
<p> and see whether I can extract all usage of YUI and do "vanilla js" submits. </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a></p>
<blockquote><p>I need a patched FF version</p></blockquote>
<p>Not quite. All you need is a Chrome that claims to be FF, then that will encounter the same problem as patched/future Firefox would. Probably simpler to get started <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler">tscherler</a>:</i>
<p><a href="https://github.com/scherler/jenkins/tree/JENKINS-53462" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/scherler/jenkins/tree/JENKINS-53462</a> I created a callback and in case we have a submit type we look up the parent form and submit it. </p>

<p>...I know but those 9 line of code is based on different approaches and to not break the universe I implemented like as-is.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler">tscherler</a>:</i>
<p><a href="https://github.com/jenkinsci/jenkins/pull/3689" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3689</a> the hack version</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Jenkins PR build for download at <a href="https://ci.jenkins.io/job/Core/job/jenkins/job/PR-3689/2/artifact/org/jenkins-ci/main/jenkins-war/2.147-rc27437.1590be3ff083/jenkins-war-2.147-rc27437.1590be3ff083.war" class="external-link" target="_blank" rel="nofollow noopener">https://ci.jenkins.io/job/Core/job/jenkins/job/PR-3689/2/artifact/org/jenkins-ci/main/jenkins-war/2.147-rc27437.1590be3ff083/jenkins-war-2.147-rc27437.1590be3ff083.war</a></p>

<p>Perhaps those watchers with ability to patch and build their own Firefox could check whether form submission buttons in this Jenkins build address the problem?</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a>, I assume that without your patch saving the settings doesn't work? Also AFAIR it was mentioned that all submit buttons throughout Jenkins are affected by that. They all work that way now? If yes, that sounds great.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>Wonderful. In that case I assume you want to finish up the patch now? Is there something else you would need help with?</p>

<p>My remaining question would be if we could also get this into a LTS release once it landed on master.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler">tscherler</a>:</i>
<p>As final explanation why FF stopped working is that since we did not provide a callback to the onClick, FF correctly stops propagating and all other browser are wrong to propagate the event to a submit.</p>

<p>We fixed the f:submit however the problem may occur in other parts but the solution will always be to provide a default fallback for onClick.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>We identified regressions that we believe are caused by this change (linked issues). This increases the likelihood that we won't choose 2.148 or newer as the next LTS baseline.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>What I wonder is that the change here should have only affected Firefox, right? But people are filing issues for Chrome and IE.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p><a href="https://github.com/jenkinsci/jenkins/pull/3689/files" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3689/files</a> is more general than that.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>Oh, I thought that this issue was about a custom behavior only used for Firefox; as stated in the issue summary. So the non-trusted `submit` event was used for all browsers then?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>:</i>
<p><a href="https://issues.jenkins.io/browse/JENKINS-54233" title="Pipeline Script textarea missing" class="issue-link" data-issue-key="JENKINS-54233">JENKINS-54233</a> was recently reported, though I cannot reproduce it and it does not seem to have anything to do with clicking buttons (though it does look like a JavaScript issue).</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Proposed reversal of the fix in <a href="https://github.com/jenkinsci/jenkins/pull/3760" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3760</a> due to regressions it caused (especially for the upcoming 2.150.x LTS line).</p>

<p>Proposed amendment of the fix in <a href="https://github.com/jenkinsci/jenkins/pull/3761" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/pull/3761</a> in the hopes it would solve the problem. Due to time constraints this has undergone minimal testing, would appreciate if others could take a look.<br/>
<a href="https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/jenkins-war/2.152-rc27510.acc4eec688a0/" class="external-link" target="_blank" rel="nofollow noopener">https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/jenkins-war/2.152-rc27510.acc4eec688a0/</a> has the jenkins.war for this PR, which currently is 2.151 + the specific change.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>Thank you for the update Daniel! I also asked on <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1399783" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1399783</a> if someone could help testing it with Firefox. Hope the new proposed fix will be less regression-prone and will fix it.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Next attempt at fixing this is done towards the next weekly release, 2.173.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>! These are great news. I cannot await to see this being shipped with that release, and for a LTS later. Is there a place I can watch in which LTS release it will be included if stable?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>2.164.3, scheduled for May 8, the earliest, if accepted as a backport. The next .1, scheduled for early June, otherwise. Assuming of course we don't cause regressions again.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza">olivergondza</a>:</i>
<p>Cool, thanks! I am backporting this to 2.164.3 to be released <del>2019-04-08</del> (EDIT: 2019-05-08), unless further regressions are found.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a>, I do appreciate the effort put in making sure Jenkins is ready for the fix in Firefox. Though I am wondering if there is a chance to postpone the delivery until significant Jenkins (or what other applications are affected) userbase migrate to the fixed version.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>I assume you mean 2019-05-08. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>We can post-pone for sure. It's not that this patch is critical, and we already waited a while for it. Do you have some stats in how long it takes for the majority of users to upgrade to the next LTS release? Will it be one week, or two? Also note that the fix will land on mozilla-central only, which means it will only be part of a nightly build. Those people are tech-aware and should upgrade their instances fast enough.</p>

<p> </p>

<p>Our next merge to beta is on 2019-05-13, so I would suggest to at least wait until 2019-05-14 before landing that change in Firefox Nightly (69).</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> thank you for bringing this up. It's actually not my job to make such a decision, so I will leave it up to the release managers to decide. I will clearly reference your last comment, so that they are aware of the situation for Jenkins users.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Reopened because 2.202 backed out the change again due to the regression it caused (<a href="https://issues.jenkins.io/browse/JENKINS-58296" title="Projects are triggered twice when using File Parameter and Firefox 67" class="issue-link" data-issue-key="JENKINS-58296"><del>JENKINS-58296</del></a>).</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>Hi <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>. After a year has been passed-by I wanted to check back if there has been made some progress on this particular issue. As known the change got backed out for 2.202, but was there any attempt again to get a fix landed?</p>

<p>On our side we would kinda like to remove this broken code from Firefox but are still blocked by this particular web-compat issue.</p>

<p>Thanks.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>I'm confused, according to <a href="https://issues.jenkins-ci.org/browse/JENKINS-58296?focusedCommentId=376769&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-376769" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-58296?focusedCommentId=376769&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-376769</a> the Firefox change was backed out again, so I thought you'd given up on this.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>You are right, sorry. I will try to get a clarification from our DOM folks. I will report back when there are news.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p>Please see <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1370630#c72" class="external-link" target="_blank" rel="nofollow noopener">https://bugzilla.mozilla.org/show_bug.cgi?id=1370630#c72</a> for a possible strategy on our side. There is no ETA yet.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>, I would like to give a brief update. With Firefox 87 we shipped a fix for that behind the Firefox preference `<tt>dom.forms.submit.trusted_event_only</tt>`. By default this preference is `false`. And there is also a deprecation warning to indicate that untrusted submit events will be removed in a future release.</p>

<p>Would that help Jenkins to fix the remaining issues, and get prepared for the official switch in Firefox at some point?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo" class="user-hover" rel="whimboo">whimboo</a> Thanks for letting us know!</p>
<blockquote><p>Form submission via untrusted submit event is deprecated and will be removed at a future date.</p></blockquote>
<p>Just tried this. Nice <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
<blockquote><p>Would that help Jenkins to fix the remaining issues, and get prepared for the official switch in Firefox at some point?</p></blockquote>
<p>I would have to look into what the current status is, but I think the previous fix caused problems with file uploads for us; those forms ended up submitted twice. The fix for that was reverting my earlier change <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>I'm not sure I understand the space well enough to fix it myself, but now that we're on notice, I hope we'll be able to get someone to look into this.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=whimboo">whimboo</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> I have seen that the <a href="https://github.com/jenkinsci/jenkins/pull/5405" class="external-link" target="_blank" rel="nofollow noopener">PR on Github</a> has been merged, which is great to see. What else is left on the Jenkins side, and what does it mean to this issue report? Can it be closed now?</p>

<p>Which version including the LTS will contain this fix? Depending on the other issues that we have to fix what is the time frame of shipping this code? Thanks.</p>
